tests/shouldwork/Cores/Xilinx/Floating/TH.hs

```diff
@@ -5,33 +5,68 @@ License    :  BSD2 (see the file LICENSE)
 Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
 -}
 
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE ViewPatterns #-}
 
 module Floating.TH where
 
-import Clash.Prelude (natToNum, unpack, Signed, Unsigned)
+import Clash.Prelude
+  ( Signed, Unsigned, BitVector, (++#), ShowX(..), BitPack
+  , msb, testBit, truncateB, showsPrecXWith, natToNum, unpack, pack
+  )
 
 import Prelude
+import Numeric (showHex)
 import Numeric.IEEE
-  (epsilon, infinity, maxFinite, minDenormal, minNormal)
+  (epsilon, infinity, maxFinite, minDenormal, minNormal, nan)
```

Comment:
`nan` is now unused as well, you can leave this change out (which will also take care of the `-Wname-shadowing`)

Reply:
I've enable `-Wall -Werror` for both files touched in this PR. Should be good now!